scripts/generators/typescript.js

```diff
@@ -0,0 +1,155 @@
+"use strict";
+
+const t = require("../../packages/babel-types");
+const utils = require("./utils");
+
+const NODE_PREFIX = "";
+
+let code = `// NOTE: This file is autogenerated. Do not modify.
+// See scripts/generators/typescript.js for script used.
+
+export class ${NODE_PREFIX}Comment {
```

Comment:
Please don't use `class`. This means  that `typeof Comment` is a constructor, that you can legitimately call `new Comment` and it will produce an object with those properties, and that you can do `instanceof Comment` to check if something is assignable to a `Comment`.

These all should be `interface`s.

Reply:
There is no valid `Comment` without a `type`, so a better definition would be something like:

```ts
interface Comment<T extends string> {
  type: T
  value: string
  ...
}

export interface BlockComment extends Comment<"BlockComment"> {}
export interface LineComment extends Comment<"LineComment"> {}
```